File: CourseProjects/CourseProjects/VisualStudioProjects/DesignPatternsExamples/Visitor/Document.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Visitor {
    internal class Document : IVisited {
        private readonly List<Module> modules = new List<Module>();

        public void AcceptVisitor(IVisitor v) {
            v.VisitDocumentStart(this);
            foreach (var m in modules) {
                m.AcceptVisitor(v);
            }
            v.VisitDocumentEnd();
        }

        public void AddModule(Module m) {
            modules.Add(m);
        }
    }
}

File: CourseProjects/CourseProjects/JavaEclipseProjects/DesignPatternsExamples/src/patterns/visitor/Document.java
package patterns.visitor;

import java.util.*;

public class Document implements Visited {
	private List<Module> modules = new ArrayList<Module>();
	
	public void acceptVisitor(Visitor v) {
		v.visitDocumentStart(this);
		for(Module m : modules) {
			m.acceptVisitor(v);
		}
		v.visitDocumentEnd();
	}
	public Document() {
		super();
	}
	public void addModule(Module m) {
		modules.add(m);
	}
}
